TESTS/netsocket/dns/main.cpp

```diff
@@ -161,7 +162,18 @@ static void net_bringup()
     net = NetworkInterface::get_default_instance();
     nsapi_error_t err = net->connect();
     TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, err);
-    printf("MBED: IP address is '%s'\n", net->get_ip_address() ? net->get_ip_address() : "null");
+    const char *address = net->get_ip_address();
+
+    printf("MBED: IP address is '%s'\n", address ? address : "null");
+#if MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE == MESH
+    uint8_t addr_hex[16];
+    stoip6(address, strlen(address), addr_hex);
+    if ((addr_hex[0] & 0xe0) != 0x20) {
+        printf("Adress is not globlal  Please check Border Router  TEST SKIP \n");
+        GREENTEA_TESTSUITE_RESULT(false);
+    }
+#endif
```

Comment:
This same code block has now been copied on many places. Can it me just one function?

Reply:
Perhaps we could follow the way this is done with a function+macro for [SKIP_IF_TCP_UNSUPPORTED](https://github.com/ARMmbed/mbed-os/blob/1798c246cc5c9f66d2c3d37dcb1c6ebb760cff9a/TESTS/netsocket/tcp/tcp_tests.h#L31)?

Reply:
Actually, @tymoteuszblochmobica pointed out to me, that this is already only copied once per test directory. We can wrap it with a function, but we'll have to  declare and define this function for every test directory (tcp, tls, dns, udp) anyway. We've been through similar discussion in https://github.com/ARMmbed/mbed-os/pull/10037 and we agreed that there is no function sharing between different test suites, unfortunately.

So I'm afraid this cannot be just one function.